### DISCUSSION TITLE
732

### DISCUSSION THREAD

**Discussion by giscus** [jojoldu/blog-comments#911] [Comments]
# 732

테스트 메소드 (함수) 이름은 비즈니스 내용을 담아야한다. 이는 테스트의 의도를 명시적으로 표현하기 때문에 중요한데, 테스트 코드를 작성하는 것에 집중한 나머지 이름에 대해서는 크게 신경쓰지 않고 넘어가는 경우가 많다. 예를 들어 다음과 같은 기존의 테스트 코드가 있다고 가정해보자. // bad describe('ArticleService', () => { it('create article pending status', async () => { const limitOverUserId = await createLimitOverUser(); const article = await sut.create(limitOverUserId, '테스트글'); expect(article.status).toBe(Article..

https://jojoldu.tistory.com/732

**Comment by kchung1995** [comment]
테스트 함수명을 비지니스 로직으로 작성하는 것은, 테스트 코드의 가독성도 높여 주지만 주어진 비지니스 로직의 요구사항에 대한 이해에도 도움을 주는 것 같아요. 머리 비우고 코드부터 작성하다 보면 요구사항이 머릿속에서 뒤섞일 때도 많더라구요 ㅎㅎ
그 외에는... 적절한 주석과 함께 사용하는 것이 테스트 코드에서는 포인트 같아요.
given-when-then 주석으로 구분하는 방법도 있고요 ㅎㅎ
좋은 글 감사합니다!
